### ISSUE TITLE
scaffold-torchtune silently uses _stable recipe even when run_val_every_n_steps is set (no validation runs)

### ISSUE THREAD

**Issue by msalganik** [niznik-dev/cruijff_kit#471]
## Summary

`scaffold-torchtune` accepts `run_val_every_n_steps` and `dataset_val` from `experiment_summary.yaml` and passes them through into `setup_finetune.yaml` / `finetune.yaml`. But the SLURM script it generates calls `cruijff_kit.tools.torchtune.custom_recipes.lora_finetune_single_device_stable.__main__`, which has **no validation logic at all** — those config keys are silently ignored.

The result: the user designed an experiment that was supposed to log mid-training validation loss every 400 steps; the experiment ran to completion without errors; and there is zero validation data anywhere. The mistake only becomes visible at analysis time when you go looking for the val curve and it isn't there.

## Where

- `scaffold-torchtune` (subagent) writes `tune run cruijff_kit.tools.torchtune.custom_recipes.lora_finetune_single_device_stable.__main__` into `finetune.slurm`.
- `src/tools/torchtune/custom_recipes/lora_finetune_single_device_stable.py` — does **not** reference `run_val_every_n_steps`, `dataset_val`, or anything val-related (`grep -nE 'val|VAL|validation' ...` only matches LoRA-validation utilities, not training-loop validation).
- `src/tools/torchtune/custom_recipes/lora_finetune_single_device_nightly.py:202+` — does implement `run_val_every_n_steps`, `dataset_val`, and logs `Validation loss: ...` to stdout + the metric logger.

## Reproducer

In `experiment_summary.yaml`:

```yaml
controls:
  validation_during_training: true
  run_val_every_n_steps: 400
data:
  training:
    splits: { train: 3000, validation: 5000, test: 0 }
```

Then `scaffold-experiment` → `run-experiment`. Training completes successfully, but `grep 'Validation loss' .../slurm-*.out` returns 0 hits and there are no `val_loss` entries in the wandb run. The `_stable` recipe took the config and threw the val keys away.

## Why it matters

For any experiment that relies on val-loss trajectories to assess plateaus, train/val gap, or early stopping, this silently turns the experiment into "train-only" — so:
- Plots of train/val loss overlay or train-val gap can't be generated.
- The setup `validation_during_training: true` is dishonest in `experiment_summary.yaml` (it claims something is enabled that isn't).
- For a binary task, val-only-via-inspect-ai-match scoring rounds to {0,1}, hiding any probabilistic information about how close the val output is to the right answer — exactly what mid-training val loss would have surfaced.

This bit me on `lora_rank_sweep_3B_parity_2026-05-06` — the analysis report has a whole "Important caveat — no mid-training validation loss" section explaining why goals (3) and (4) of the original experiment design can't be delivered.

## Suggested fix

Pick one of:

1. **Fail fast at scaffold time.** When `controls.run_val_every_n_steps` (or `validation_during_training: true`, or `dataset_val`) is set in `experiment_summary.yaml`, `scaffold-torchtune` should refuse to use the `_stable` recipe and either error out or auto-switch to `_nightly`. A loud warning is the minimum; a hard error is more honest.
2. **Backport val support to `_stable`.** If `_stable` is meant to be the default recipe (and `_nightly` is more experimental), the validation loop should be merged in. The `_nightly` implementation at lines 200–860 is small; it's not a deep change.
3. **Add a `recipe:` field to experiment_summary.yaml controls** and have scaffold-torchtune pick `_stable` vs `_nightly` from there explicitly, with the schema validator complaining when val keys are set against `_stable`.

Option 1 is the cheapest fix and prevents the silent-failure mode going forward. Option 2 is the right long-term fix.

## Discovered while

Analyzing the `lora_rank_sweep_3B_parity_2026-05-06` experiment in the `model_organism` project. See `analysis/report.md` in that experiment for the downstream analysis impact (whole categories of plots had to be dropped from the report).

**Comment by niznik-dev** [comment]
We should be defaulting to nightly at this point, not sure why it chose stable here

**Comment by msalganik** [comment]
That's what I thought.  You can find the "brief" I used here: /scratch/gpfs/MSALGANIK/mjs3/ck-projects/model_organism/lora_rank_sweep_3B_parity_2026-05-06/briefs

Not sure if you would see the same behavior or not.  Could be something on my machine?  Might be a good example for a future test?
